src/code/z_kankyo.c

```diff
@@ -732,13 +1682,123 @@ void func_800F8970(void) {
     }
 }
 
+#ifdef NON_MATCHING
+void Environment_UpdatePostmanEvents(PlayState* play) {
+    u8 v1;
+    u16 temp_a2_2;
+
+    if ((u8)((void)0, ((gSaveContext.eventInf[7] & 0xE0) >> 5)) != (u8)((void)0, gSaveContext.save.day)) {
+        v1 = ((void)0, (gSaveContext.eventInf[7] & (u8)~0xE0)) | (((void)0, gSaveContext.save.day) << 5);
+        gSaveContext.eventInf[7] = v1;
+        SET_WEEKEVENTREG(WEEKEVENTREG_27_40);
+        SET_WEEKEVENTREG(WEEKEVENTREG_27_80);
+        SET_WEEKEVENTREG(WEEKEVENTREG_28_01);
+        SET_WEEKEVENTREG(WEEKEVENTREG_28_02);
+        SET_WEEKEVENTREG(WEEKEVENTREG_28_04);
+
+        func_800F88C4(WEEKEVENTREG_DEPOSITED_LETTER_TO_KAFEI_SOUTH_UPPER_CLOCKTOWN);
+        func_800F88C4(WEEKEVENTREG_DEPOSITED_LETTER_TO_KAFEI_NORTH_CLOCKTOWN);
+        func_800F88C4(WEEKEVENTREG_DEPOSITED_LETTER_TO_KAFEI_EAST_UPPER_CLOCKTOWN);
+        func_800F88C4(WEEKEVENTREG_DEPOSITED_LETTER_TO_KAFEI_EAST_LOWER_CLOCKTOWN);
+        func_800F88C4(WEEKEVENTREG_DEPOSITED_LETTER_TO_KAFEI_SOUTH_LOWER_CLOCKTOWN);
+
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_27_40);
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_27_80);
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_28_01);
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_28_02);
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_28_04);
+    }
+
+    if ((SCHEDULE_TIME_NOW < 0x1FFE) || (SCHEDULE_TIME_NOW >= 0x3FFD)) {
+        CLEAR_WEEKEVENTREG(WEEKEVENTREG_90_08);
+    }
+
+    func_800F8970();
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_ROOM_KEY) && !CHECK_WEEKEVENTREG(WEEKEVENTREG_55_02)) {
+        if (((void)0, gSaveContext.save.day) >= 2) {
+            SET_WEEKEVENTREG(WEEKEVENTREG_55_02);
+        } else if ((((void)0, gSaveContext.save.day) == 1) && (SCHEDULE_TIME_NOW >= 0x6FF9)) {
+            SET_WEEKEVENTREG(WEEKEVENTREG_55_02);
+        }
+    }
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_90_01)) {
+        temp_a2_2 = ((void)0, gSaveContext.save.time) - D_801F4E78;
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_89_40) && ((SCHEDULE_TIME_NOW & 0xFFFF) >= 0xF556)) {
```

Comment:
```suggestion
        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_89_40) && ((u16)SCHEDULE_TIME_NOW >= 0xF556)) {
```
I'd argue for a `u16` cast instead.